Firebase/Database/Login/FAuthTokenProvider.m

```diff
@@ -16,65 +16,50 @@
 
 #import "FAuthTokenProvider.h"
 #import "FUtilities.h"
+#import "FIRAppInternal.h"
 #import "FIRLogger.h"
 #import "FIRDatabaseQuery_Private.h"
 #import "FIRNoopAuthTokenProvider.h"
 
-static NSString *const FIRAuthStateDidChangeInternalNotification = @"FIRAuthStateDidChangeInternalNotification";
-static NSString *const FIRAuthStateDidChangeInternalNotificationTokenKey = @"FIRAuthStateDidChangeInternalNotificationTokenKey";
-
-
-/**
- * This is a hack that defines all the methods we need from FIRFirebaseApp. At runtime we use reflection to get an
- * actual instance of FIRFirebaseApp. Since protocols don't carry any runtime information and selectors are invoked
- * by name we can write code against this protocol as long as the method signatures of FIRFirebaseApp don't change.
- */
-@protocol FIRFirebaseAppLike <NSObject>
-
-- (void)getTokenForcingRefresh:(BOOL)forceRefresh withCallback:(void (^)(NSString *_Nullable token, NSError *_Nullable error))callback;
-
-@end
-
-
 /**
  * This is a hack that defines all the methods we need from FIRAuth.
  */
 @protocol FIRFirebaseAuthLike <NSObject>
 
-- (id<FIRFirebaseAppLike>) app;
+- (FIRApp *) app;
 
 @end
 
 /**
  * This is a hack that copies the definitions of Firebase Auth error codes. If the error codes change in the original code, this
  * will break at runtime due to undefined behavior!
  */
-typedef NS_ENUM(NSUInteger, FIRErrorCode) {
-    /*! @var FIRErrorCodeNoAuth
+typedef NS_ENUM(NSUInteger, FIRMockErrorCode) {
```

Comment:
Hrm. This is another FIRApp hack that we should ideally remove.  The consumer can't tell if [FIRApp getTokenForcingRefresh:] actually tried to get a token and failed (e.g. no network connection so we can't talk to the backend) or if the app just doesn't use auth (so we should proceed unauthenticated), or because the app uses auth but no user is currently signed in (so again we should proceed unauthenticated).

Any chance we can tackle this too (move these errors into core perhaps) here or in a subsequent PR?